src/NServiceBus.Transport.RabbitMQ/Connection/ConnectionFactory.cs

```diff
@@ -103,7 +107,15 @@ public IConnection CreateConnection(string connectionName, bool automaticRecover
                 connectionFactory.AutomaticRecoveryEnabled = automaticRecoveryEnabled;
                 connectionFactory.ConsumerDispatchConcurrency = consumerDispatchConcurrency;
 
-                var connection = connectionFactory.CreateConnection(connectionName);
+                IConnection connection = null;
+                if (additionalHostnames != null && additionalHostnames.Count > 0)
+                {
+                    connection = connectionFactory.CreateConnection(additionalHostnames, connectionName);
```

Comment:
shouldn't this also take the "primary" host name into account?

Reply:
:+1: Good catch! I've pushed a fix